### PR TITLE
LPS-152585 [CP 2.0] - [Impedibug] Disable 'Generate X Key' button and add description

### PR DESCRIPTION
**Pull request by claricebrasil**
Fix enable generate key button without filling one or more Server Details and add a info description to show that the user has to fill at least one field.

### CODE REVIEW

**Review by thaynam — APPROVED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/SelectSubscription.js [hunk 1]
```diff
@@ -360,7 +360,11 @@ const SelectSubscription = ({
 										setInfoSelectedKey(infoSelectedKey);
 									}}
 									selected={selected}
-									subtitle={`Instance size: ${subscriptionTerm.instanceSize}`}
+									subtitle={`Instance size: ${
+										subscriptionTerm.instanceSize === 0
+											? subscriptionTerm.instanceSize + 1
+											: subscriptionTerm.instanceSize
+									}`}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/SelectSubscription.js:367]:
```suggestion
										subscriptionTerm?.instanceSize || 1
									}`}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.js [hunk 2]
```diff
@@ -114,6 +116,13 @@ const RequiredInformation = ({
 		}
 		else {
 			const productName = `${infoSelectedKey?.productType} ${infoSelectedKey?.licenseEntryType}`;
+			const sizing =
+				infoSelectedKey?.selectedSubscription.instanceSize === 0
+					? `Sizing ${
+							infoSelectedKey?.selectedSubscription
+								?.instanceSize + 1
+					  }`
+					: `Sizing ${infoSelectedKey?.selectedSubscription.instanceSize}`;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.js:125]:
```suggestion
			const sizing = `Sizing ${ infoSelectedKey?.selectedSubscription?.instanceSize || 1 }`
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/GenerateCardLayout/index.js [hunk 3]
```diff
@@ -58,10 +58,12 @@ const GenerateCardLayout = ({infoSelectedKey}) => {
...
-							{
-								infoSelectedKey?.selectedSubscription
-									?.instanceSize
-							}
+							{infoSelectedKey?.selectedSubscription
+								?.instanceSize === 0
+								? infoSelectedKey?.selectedSubscription
+										?.instanceSize + 1
+								: infoSelectedKey?.selectedSubscription
+										?.instanceSize}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/GenerateCardLayout/index.js:66]:
```suggestion
							{infoSelectedKey?.selectedSubscription
								?.instanceSize || 1}
```

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/SelectSubscription.js (hunk 1 above)
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/SelectSubscription.js:367]: Nice!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.js (hunk 2 above)
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/RequiredInformation.js:125]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/GenerateCardLayout/index.js (hunk 3 above)
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/GenerateCardLayout/index.js:66]: Done!

**Pull request closed without merging**